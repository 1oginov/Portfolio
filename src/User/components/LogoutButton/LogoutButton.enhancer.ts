import { withFirebase } from 'react-redux-firebase';
import { compose, withHandlers } from 'recompose';

import { WithFirebaseHocProps } from 'Firebase/lib';

import { Props } from './LogoutButton';

export default compose<Props & WithFirebaseHocProps, Partial<Props>>(
  withFirebase,
  withHandlers<WithFirebaseHocProps, {}>({

    handleClick: ({ firebase }) => () => {
      firebase.auth().signOut();
    },

  }),
);
