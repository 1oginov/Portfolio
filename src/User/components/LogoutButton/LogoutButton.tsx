import { Button } from '@material-ui/core';
import * as React from 'react';

export interface Props {
  children: React.ReactNode;
  className?: string;
  handleClick: () => void;
}

const LogoutButton: React.FunctionComponent<Props> = ({ children, className, handleClick }: Props) => (
  <Button className={className} onClick={handleClick} variant="outlined">{children}</Button>
);

export default LogoutButton;
