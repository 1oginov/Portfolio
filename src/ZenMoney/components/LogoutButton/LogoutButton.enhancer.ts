import { connect } from 'react-redux';
import { compose, withHandlers } from 'recompose';

import { logout as logoutAction } from '../../actions';

const mapDispatchToProps = { logout: logoutAction };

export default compose(
  connect(null, mapDispatchToProps),
  withHandlers({

    handleClick: ({ logout }) => () => {
      logout();
    },

  }),
);
