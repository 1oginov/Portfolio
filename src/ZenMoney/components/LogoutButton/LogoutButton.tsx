import * as React from 'react';

const LogoutButton = ({ handleClick }) => (
  <button onClick={handleClick}>Logout</button>
);

export default LogoutButton;
