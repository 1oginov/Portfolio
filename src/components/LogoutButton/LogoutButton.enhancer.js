/* @flow */

import { connect } from 'react-redux';
import { compose, withHandlers } from 'recompose';

import { logout } from '../../actions';

const mapDispatchToProps = {
  logout,
};

export default compose(
  connect(null, mapDispatchToProps),
  withHandlers({

    handleClick: ({ logout }) => () => {
      logout();
    },

  }),
);
