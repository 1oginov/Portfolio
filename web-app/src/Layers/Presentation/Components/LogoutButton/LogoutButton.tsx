import { Button } from '@material-ui/core';
import * as React from 'react';

import {
  LogoutButtonEnhancerInputProps, LogoutButtonEnhancerProps,
} from 'Layers/Behavior/Enhancers/LogoutButtonEnhancer/LogoutButtonEnhancer';

export interface LogoutButtonProps extends LogoutButtonEnhancerInputProps {
  children: React.ReactNode;
  className?: string;
}

type Props = LogoutButtonProps & LogoutButtonEnhancerProps;

export const LogoutButton: React.FunctionComponent<Props> = ({ children, className, handleClick }: Props) => (
  <Button className={className} onClick={handleClick} variant="outlined">{children}</Button>
);
