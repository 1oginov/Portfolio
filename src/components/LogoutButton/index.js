/* @flow */

import LogoutButton from './LogoutButton';
import enhance from './LogoutButton.enhancer';

export default enhance(LogoutButton);
