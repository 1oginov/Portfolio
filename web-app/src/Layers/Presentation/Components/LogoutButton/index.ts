import { LogoutButtonEnhancer } from 'Layers/Behavior/Enhancers/LogoutButtonEnhancer/LogoutButtonEnhancer';

import { LogoutButton as LogoutButtonComponent, LogoutButtonProps } from './LogoutButton';

export const LogoutButton = LogoutButtonEnhancer<LogoutButtonProps>()(LogoutButtonComponent);
