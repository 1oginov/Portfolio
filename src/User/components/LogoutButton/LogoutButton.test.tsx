import { shallow } from 'enzyme';
import * as React from 'react';

import LogoutButton from './LogoutButton';

it('matches snapshot', () => {
  const wrapper = shallow(<LogoutButton handleClick={() => undefined}>Logout</LogoutButton>);

  expect(wrapper).toMatchSnapshot();
});
