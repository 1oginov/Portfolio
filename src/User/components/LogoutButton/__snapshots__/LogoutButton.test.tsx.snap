// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<WithStyles(ForwardRef(Button))
  onClick={[Function]}
  variant="outlined"
>
  Logout
</WithStyles(ForwardRef(Button))>
`;
